src/main.rs

```diff
@@ -51,7 +51,8 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     };
     let _ = CombinedLogger::init(loggers);
 
-    let _ = connection::read_stream_key(true);
+    let stream_key_env = matches.value_of("stream-key");
+    let _ = connection::read_stream_key(true, stream_key_env);
```

Comment:
Hm okay I see what you are doing here but I am not sure this is the most elegant solution. Couldn't we just read the value of the ENV var on startup and remove the need for the additional argument in the function?

Reply:
Okay I see why you did it that way now. Id still like to try and find a better solution